File: src/Knapsack.java
/*
 * Please find below DP solution. 
 * Complexity of the algorithm is O(m*n)
 * where m is capacity and the n is weights.
 */
public class Knapsack {
	// "static void main" must be defined in a public class.

	public static void main(String args[]) {
		int val[] = new int[] { 60, 100, 120 };
		int wt[] = new int[] { 10, 20, 30 };
		int W = 50;
		int n = val.length;
		System.out.println(knapSack(W, wt, val, n));
	}

	private static int knapSack(int w, int[] wt, int[] val, int n) {

		int finalAmount = 0;

		int[][] dp = new int[n + 1][w + 1];

		for (int i = 1; i < dp.length; i++) {

			for (int j = 1; j < dp[0].length; j++) {

				if (j < wt[i - 1]) {
					dp[i][j] = dp[i - 1][j];
				} else {
					dp[i][j] = Math.max(dp[i - 1][j - wt[i - 1]] + val[i - 1], dp[i - 1][j]);
				}
			}
		}

		return dp[dp.length - 1][dp[0].length - 1];

	}
}

File: knapsack.swift
func MaxCapacity (_ values : [Int], _ weights: [Int], _ capacity: Int) -> Int {
        var dp = Array(repeating: Array(repeating: 0, count: capacity + 1), count : values.count + 1)
        for i in 1...values.count {
            for j in 1...capacity {
                if j < weights[i-1] {
                    dp[i][j] = dp[i-1][j]
                } else {
                    dp[i][j] = max(dp[i-1][j],dp[i-1][j-weights[i-1]] + values[i-1])
                }
            }
        }
        return dp[values.count][capacity]
    }
print(MaxCapacity([10,6,12],[2,1,3],5))